### PR TITLE
haxe.macro.type documentation

### PR DESCRIPTION
**Pull request by markknol**
I found it quite hard to describe everything in these modules, but It's nice to have at least a simple description to everything in the macro context, it should be better than having nothing. 

Some questions:
* when is `TWhile`'s `normalWhile` true/false?
* what is `TUnop` and `TBinop`
* how to describe `Ref<T>`

### CODE REVIEW

**Comment by nadako** [comment]
* I think when `normalWhile` is true, it's `while (...) {...}`, and if it's false - it's `do {...} while (...)`.
 * `TUnop` is unary operation, `TBinop` is binary :-)
 * `Ref<T>` represents a reference to internal compiler structure. It exists because encoding compiler structures for use in macro is quite expensive, so this is kind of "lazy" encoding - a structure is only encoded when user requests it through `ref.get()`.

**Comment by Gama11** [comment]
@nadako The Manual actually uses "metatag" in some places:

http://haxe.org/manual/target-javascript-metatags.html
http://haxe.org/manual/target-flash-metatags.html

Most of the time it uses "metadata" though. Perhaps that should be made consistent?

**Comment by Simn** [comment]
It should be metadata, not metatag.

**Comment by markknol** [comment]
It is metadata in the comment line.

**Comment by markknol** [comment]
- I have TCall in
- I have "in macros" in too

Good to go?

File: std/haxe/macro/Type.hx [hunk 1]
```diff
@@ -312,36 +375,128 @@ enum FieldAccess {
...
+	/**
+		Represents an object declaration.
+	**/
 	TObjectDecl(fields:Array<{name:String, expr:TypedExpr}>);
+	/**
+		Represents an array declaration.
+	**/
 	TArrayDecl(el:Array<TypedExpr>);
+	/**
+		Represents a function call declaration.
```
Comment (Simn) [inline std/haxe/macro/Type.hx]: It's not a declaration of any sort, it's just a function call.

**Comment by markknol** [comment]
Is this good to go?

**Comment by Simn** [comment]
I want to make some changes, but I'll do that myself.

**Comment by Simn** [comment]
I have reworked this and documented the missing types and fields. Two remarks for the future:

* The Haxe AST has no statements, everything is an expression
* You used the term "declaration" a lot in places where it is rather confusing (e.g. "break-statement declaration").

**Comment by markknol** [comment]
Looks great!

**Pull request closed without merging**